### ISSUE TITLE
Images behind a tansparency are replaced in front of the transparency

### ISSUE THREAD

**Issue by Mattat01** [lifelike/countersheetsextension#103]
I understand that the extension is doing a search and replace to insert the right image on each counter. In my case I have a card with some of the text behind a transparency (75% white) to dull that region of the card. Any text or image remain under the transparency and come out fine, but inline images must be placed at a higher (top?) layer and so is is on top of the white transparency. 

It seems to occur as the inline images are not in the same group as the original text field, but all inline images are in a group that also contains the rest of the card's group. I can fix it by manually dragging the image into the card's group, at a layer level above or below the orignal text (but below the transparency).

**Comment by lifelike** [comment]
Yes, the images are inserted at the top of the group, not just on top of the text. Thanks for reporting this. Will find some time asap to have a closer look.

**Comment by Mattat01** [comment]
Thank you so much. I can't help finding how to fix it (my Python isn't good enough), but perhaps this helps save you some time:

It isn't a bug, clearly the method coded works in most cases. 

I see that most substitutions append the results back to the originating text node/element, such as insertImagePlaceholder() where the placeholder 'X' is added. Inserting this placeholder originates back in the generatecounter function (line 1028 calls substitute_text, and it goes through from there) and obviously these replacements are occuring on/in the correct layer. 

But then the placeholder is 'replaced/covered' from line 1945, and ultimately added back to the top level group (on line 1984 "group.append(image)"). The element containin the 'X' placeholder is available in the placeholders array, but I don't know if you can get back to the placeholder's layer reference or not, so that the image can be appended to the layer's parent immediately before the placeholder element instead of just to the top level group.

I'm sure you know your (excellent) code much better than me, I'm just trying to do what I can to save you a few minutes :)

Thanks.
